### PR TITLE
[ETCM-141] scalafmt

### PR DESCRIPTION
**Pull request by lemastero**
# Description

Add [scalafmt plugin](https://github.com/scalameta/sbt-scalafmt) and format all code.

### CODE REVIEW

**Comment by rtkaczyk** [comment]
Should we add `scalafmtCheckAll` to buildkite?

File: .buildkite/pipeline.nix [hunk 1]
```diff
@@ -54,6 +54,14 @@ in
       '';
     };
 
+    scalafmt = commonAttrs // {
+      dependsOn = [ compile ];
```
Comment (kapke) [inline .buildkite/pipeline.nix:58]: Why not other way around? Checking formatting is much more lightweight than compiling.

Comment (lemastero) [reply .buildkite/pipeline.nix]: Nice idea, changed, and now compile depends on formatting.

**Review by kapke — APPROVED** [review]
Let's get this merged as soon as CI passes

**Pull request merged**